src/crl/mod.rs

```diff
@@ -128,6 +145,10 @@ impl<'a> RevocationOptions<'a> {
             (None, _) => return Err(Error::UnknownRevocationStatus),
         };
 
+        if self.expiration_policy == ExpirationPolicy::Enforce {
```

Comment:
This should be below the signature check -- the nextUpdate field is inside TBS and should be treated as indeterminate until the signature is validated.